### DISCUSSION TITLE
A workaround for wrapped copy-paste in Tmux

### DISCUSSION THREAD

**Discussion by amimof** [wezterm/wezterm#5798] [Show and tell]
I've been having issues copy-pasting in Tmux. The problem is that wrapped lines of text are pasted from clipboard with linebreaks. See this issue for more information https://github.com/wez/wezterm/issues/4706. I wasn't able to comment in the issue since it's closed. I apologize if this is the wrong place for this kind of information but I feel that I need to share my workaround for this.

Essentially what I did was to create a key mapping for `super+c` (I'm on Mac) to emit an event. Then I listen for that event which extracts the selected text and removes any line breaks before committing it back to clipboard.

```lua
wezterm.on('log-selection', function(window, pane)
  local sel = window:get_selection_text_for_pane(pane)
  local str = sel:gsub("\r?\n", "")
  window:copy_to_clipboard(str, 'Clipboard')
end)

local act = wezterm.action
config.keys = {
  { key = "c", mods = "SUPER", action = act.EmitEvent 'log-selection' }
}

```

Hope this helps. Cheers

**Comment by govardha** [comment]
Hello @amimof 

Thanks for posting this, I tried using this but to no avail, can you please correct me with what I am doing.

I am using the following plugin
https://github.com/sei40kr/wez-tmux/tree/main

Steps I followed.
1.  Engage the leader and select a block of text 
2.  I am on windows so once the text is copied into the clipbod, I engage the windows + c key as you have done
3.  Finally I paste the contents of the clipboard into wherever I want to paste in and I still end up getting the new line.

Can you please post what I am doing wrong in my workflow when using your code?

Thank you!
g

**Reply by amimof** [reply]
Your workflow seems fine to me. Only difference is that I'm not using wez-tmux plugin, and I'm on MacOS. Try printing the copied `str` to console and see what you get. For example with:

```
wezterm.log_info("str: " .. str)
```

**Reply by wez** [reply]
FWIW, support for `tmux -CC` was very recently merged into nightly builds of wezterm. That causes tmux's windows and panes to be mapped to wezterm tabs and panes and should solve this issue.
